lib/archethic/contracts/interpreter/library/common/chain.ex

```diff
@@ -3,22 +3,24 @@ defmodule Archethic.Contracts.Interpreter.Library.Common.Chain do
   @behaviour Archethic.Contracts.Interpreter.Library
 
   alias Archethic.Contracts.Interpreter.ASTHelper, as: AST
-  alias Archethic.Contracts.Interpreter.Legacy
-
-  @spec get_genesis_address(binary()) :: binary()
-  defdelegate get_genesis_address(address),
-    to: Legacy.Library,
-    as: :get_genesis_address
-
-  @spec get_first_transaction_address(binary()) :: binary()
-  defdelegate get_first_transaction_address(address),
-    to: Legacy.Library,
-    as: :get_first_transaction_address
-
-  @spec get_genesis_public_key(binary()) :: binary()
-  defdelegate get_genesis_public_key(public_key),
-    to: Legacy.Library,
-    as: :get_genesis_public_key
+
+  @callback get_genesis_address(binary()) :: binary()
+  @callback get_first_transaction_address(binary()) :: binary()
+  @callback get_genesis_public_key(binary()) :: binary()
+
+  def get_genesis_address(address), do: impl().get_genesis_address(address)
```

Comment:
You can use knigge to remove the boilerplate of functions dedicated to the default impl